object-cache.php

```diff
@@ -439,6 +445,25 @@ function get( $id, $group = 'default', $force = false, &$found = null ) {
 			if ( false === $flags ) {
 				$found = false;
 				$value = false;
+			} elseif ( false === $value && ( $flags & 0xFF01 ) === 0x01 ) {
+				/*
+				 * The lowest byte is used for flags.
+				 * 0x01 means the value is serialized (MMC_SERIALIZED).
+				 * The second lowest indicates data type: 0 is string, 1 is bool, 3 is long, 7 is double.
+				 * `null` is serialized into a string, thus $flags is 0x0001
+				 * Empty string will correspond to $flags = 0x0000 (not serialized).
+				 * For `false` or `true` $flags will be 0x0100
+				 *
+				 * See:
+				 * - https://github.com/websupport-sk/pecl-memcache/blob/2a5de3c5d9c0bd0acbcf7e6e0b7570f15f89f55b/php7/memcache_pool.h#L61-L76 - PHP 7.x
+				 * - https://github.com/websupport-sk/pecl-memcache/blob/ccf702b14b18fce18a1863e115a7b4c964df952e/src/memcache_pool.h#L57-L76 - PHP 8.x
+				 *
+				 * In PHP 8, they changed the way memcache_get() handles `null`:
+				 * https://github.com/websupport-sk/pecl-memcache/blob/ccf702b14b18fce18a1863e115a7b4c964df952e/src/memcache.c#L2175-L2177
```

Comment:
Because this is just so weird, I dug through looking for the actual commit in the PR:
https://github.com/websupport-sk/pecl-memcache/pull/76/commits/d91338af0c493931cd49460cfc54428850ca0af5

```
- fix functions returning NULL instead of advertised return type,
  For example, when memcached returned error on "delete", function
  returned null instead of false
```

Not returning a stored `null` value doesn't look like an intentional change, rather a compat bug.

Reply:
Yeah, but unfortunately this bug is almost two years old; it is unlikely that it will be fixed.

Reply:
> Do we also care for MMC_COMPRESSED

No, we need only to find out if the value is serialized (0x??01 mask) and its type (0xFF?? mask). If value's type is `string`, then if it is serialized, the value is `null`, otherwise it is an empty string.

We can check `$flags & 0x02` to see if the value was compressed but we don't really need this information here.